app/controllers/sign_up/recovery_codes_controller.rb

```diff
@@ -35,6 +35,8 @@ def next_step
         sign_up_completed_path
       elsif current_user.password_reset_profile.present?
         reactivate_profile_path
+      elsif (stored_location = stored_location_for(current_user))
```

Comment:
not sure if this is what we want, but could also take advantage of `after_sign_in_path` method.

https://github.com/18F/identity-idp/blob/ca722eae3857744c282b29ce7509625f40f943fa/app/controllers/application_controller.rb#L60